src/modals/AddTaskModal.js

```diff
@@ -35,15 +36,25 @@ export default class AddTaskModal extends Component<{}> {
 
   constructor(props){
     super(props);
+    this.formatSelectedDate = this.formatSelectedDate.bind(this);
     this.state = {
       taskText: '',
-      date: currentDate
+      date: currentDate,
+      milliDate: ''
     }
   }
 
+  formatSelectedDate(date){
+    const dateChangeMilli = new Date( Date.parse(date) );
+    let month = monthDate[dateChangeMilli.getMonth()];
+    let dateChangeFormat = month + ' ' + dateChangeMilli.getDate();
+    this.state.milliDate = dateChangeFormat;
+  }
+
   addTask(){
+    this.formatSelectedDate(this.state.date);
     if(this.state.taskText !== ''){
-      this.props.addTask(this.state.taskText);
+      this.props.addTask(this.state.taskText, this.state.milliDate);
```

Comment:
Para que le mandar this.state.milliDate si ya tienes en el estado la fecha que el DatePicker se encarga de guardar?  Lo que estas haciendo es guardar la fecha actual con un cierto formato (no se por que aún) en lugar de la fecha que el usuario ya eligió en el DatePicker..